Flow.Launcher/ViewModel/MainViewModel.cs

```diff
@@ -708,36 +711,56 @@ private void SetOpenResultModifiers()
 
         public void ToggleFlowLauncher()
         {
-            if (MainWindowVisibility != Visibility.Visible)
+            if (WinToggleStatus != true)
             {
-                MainWindowVisibility = Visibility.Visible;
+                Show();
             }
             else
             {
                 Hide();
             }
         }
 
+        public void Show()
+        {
+            if (_settings.UseSound)
+            {
+                MediaPlayer media = new MediaPlayer();
+                media.Open(new Uri(AppDomain.CurrentDomain.BaseDirectory + "Resources\\open.wav"));
+                media.Play();
+            }
+            MainWindowVisibility = Visibility.Visible;
+            WinToggleStatus = true;
+            ((MainWindow)Application.Current.MainWindow).WindowAnimator();
+            MainWindowOpacity = 1;
+        }
+
         public async void Hide()
         {
+            MainWindowOpacity = 0;
             switch (_settings.LastQueryMode)
             {
                 case LastQueryMode.Empty:
                     ChangeQueryText(string.Empty);
-                    Application.Current.MainWindow.Opacity = 0; // Trick for no delay
-                    await Task.Delay(100);
-                    Application.Current.MainWindow.Opacity = 1;
+                    MainWindowOpacity = 0; // Trick for no delay
+                    await Task.Delay(100); //Time for change to opacity
                     break;
                 case LastQueryMode.Preserved:
+                    MainWindowOpacity = 0;
+                    await Task.Delay(100);
                     LastQuerySelected = true;
                     break;
                 case LastQueryMode.Selected:
+                    MainWindowOpacity = 0;
+                    await Task.Delay(100);
```

Comment:
are these Task.Delay(100) related to animation too?

Reply:
yep

Reply:
hmm, i am finding these task delays are causing Hide to be called twice when toggle the search window very quickly three times.
repro:
1. turn off animation
2. hide search window
3. quickly press the toggle hotkey 3 times
4. the expected behaviour should be show->hide->show, but it's coming up as show->hide->show->hide

you can use Debug.WriteLine("Hide") to write out to output.

Reply:
done. If the animation setting is off, there is no delay.